src/server/services/procedures/rain-viewer/rain-viewer.js

```diff
@@ -0,0 +1,169 @@
+const FsStore = require('cache-manager-fs');
+const utils = require('../utils');
+const jimp = require('jimp');
+const _ = require('lodash');
+const fs = require('fs');
+
+const { defineTypes, TIME_OFFSET_MAP } = require('./types');
+defineTypes();
+
+const GoogleMaps = require('../google-maps/google-maps');
+
+const CACHE_DIR = process.env.CACHE_DIR || 'cache';
+logger.trace(`cache dir (root): ${CACHE_DIR}`);
+if (!fs.existsSync(CACHE_DIR)) {
+    fs.mkdirSync(CACHE_DIR); // ensure path exists
+}
+const MapCache = CacheManager.caching({
+    store: FsStore,
+    options: {
+        ttl: Infinity,
+        maxsize: 1024 * 1000 * 100,
+        path: `${CACHE_DIR}/RainViewer`,
+        preventfill: false,
+        reviveBuffers: true,
+    },
+});
+
+const RainViewer = new ApiConsumer('RainViewer', '', { cache: { ttl: 5 * 60 } }); // radar data updates every 10 minutes (or so)
+ApiConsumer.setRequiredApiKey(RainViewer, GoogleMapsKey);
+
+function decimalize(val) {
+    const str = val.toString();
+    return str.includes('.') ? str : `${str}.0`;
+}
+
+/**
+ * Get the list of valid radar time offsets for the :func:`RainViewer.getOverlay` RPC.
+ * The returned time offsets are in chronological order.
+ *
+ * @returns {Array<TimeOffset>} The list of valid time offsets in chronological order.
+ */
+RainViewer.getTimeOffsets = function () {
+    return Object.keys(TIME_OFFSET_MAP);
+};
+
+/**
+ * Gets a transparent overlay that can be placed directly on to of a map provided by :func:`GoogleMaps.getMap`
+ * to display recent or forecasted weather radar data.
+ *
+ * If you would rather have a single image with a map already added in the background,
+ * you can specify the ``background`` optional parameter with the type of map to use.
+ *
+ * @param {Latitude} latitude Latitude of the returned map (centered).
+ * @param {Longitude} longitude Longitude of the returned map (centered).
+ * @param {BoundedInteger<1>} width Width (in pixels) of the returned map.
+ * @param {BoundedInteger<1>} height Height (in pixels) of the returned map.
+ * @param {BoundedInteger<1,25>} zoom The zoom level of the returned image (see the :doc:`/service/GoogleMaps/index` service).
+ * @param {TimeOffset=} timeOffset The time offset of the desired forecast (defaults to ``now``, which represents current weather).
+ * @param {Object=} options Additional drawing options.
+ * @param {Enum<none,roadmap,satellite,terrain>=} options.background Type of map to use for the background of the image (radar overlay on top) (default none).
+ * @param {Boolean=} options.smooth If set to true, smooths the radar overlay in the returned image to be more aesthetically pleasing (default true).
+ * @param {Boolean=} options.showSnow If set to true, renders snow as a separate color from normal precipitation (default false).
+ * @param {BoundedInteger<0,21>=} options.colorScheme An integer denoting the color scheme to use in the returned image (default 4).
```

Comment:
There are names for 0-8 (https://www.rainviewer.com/api/color-schemes.html), but the others are all unnamed and completely undocumented. I only found out they existed by trying out of bounds values to see what error it would give. I guess we could assign them names ourselves, though (probably also rename the ones they named since they wouldn't match the others).

Reply:
Also for if you try it out locally, I believe these project files were up to date: [rain-viewer-projs.zip](https://github.com/NetsBlox/NetsBlox/files/8508407/rain-viewer-projs.zip)